Fix link-checking and a number of links

dholbach opened the pull request:
- Improve link checking
- fix a number of links
- update theme

hiddeco commented on pkg/apis/helm.fluxcd.io/v1/types_helmrelease.go:
```diff
@@ -183,7 +183,7 @@ func (s GitChartSource) RefOrDefault(defaultGitRef string) string {
 type RepoChartSource struct {
 	// RepoURL is the URL of the Helm repository, e.g.
 	// https://kubernetes-charts.storage.googleapis.com or
```
Can you make all these code blocks for in-documentation prettiness?

dholbach commented:
@stefanprodan @hiddeco Not sure how to proceed here. Shall I split the PR into 1) fixed links and 2) link checking and we take some more time to figure out the broken links in the 404 document?

hiddeco commented:
Yes, please split the PR so we can take the proper time to address the CI-part.

dholbach commented:
Moved fixing the links to #356

dholbach commented:
@hiddeco @stefanprodan I'm getting closer on finishing this. To me it somehow looks like the 404.html site generation is busted.

Commit:
fix link-checking (follow fluxcd/flux#2956)

	move back to upstream GH action
	update pymdown-extensions
	update to mkdocs-material 5: https://squidfunk.github.io/mkdocs-material/releases/5/#how-to-upgrade
	fix links
	follow fluxcd/flux#3006
	parallelise checking

Commit:
try out dropping site_url

dholbach commented:
The tests pass now, but https://test-helm-op.readthedocs.io/en/latest/missing is still busted. :-(

dholbach commented:
To confirm locally:

```console
pip3 install -r docs/requirements.txt
mkdocs build -c -s && firefox site/404.html
```

dholbach commented:
I suggest merging this - it's been sitting there long enough. It's an improvement - let's take a look at the 404 separately.

hiddeco commented on CONTRIBUTING.md:
```diff
@@ -72,10 +78,10 @@ This is a rough outline of how to prepare a contribution:
 ### How to build and run the project
 
-Refer to the [Building](https://docs.fluxcd.io/projects/helm-operator/en/stable/contributing/building/) guide to find out how to build from
+Refer to the [Building](https://docs.fluxcd.io/en/stable/contributing/building/) guide to find out how to build from
```
The Helm Operator now has its own version of these types of document but they have not ended up in a `stable` release yet. I do however think it would be fine to use `latest` links here, in which case they URL would be: https://docs.fluxcd.io/projects/helm-operator/en/latest/contributing/building/

hiddeco commented on CONTRIBUTING.md:
```diff
@@ -72,10 +78,10 @@ This is a rough outline of how to prepare a contribution:
 source.
 
-Refer to the [Get started developing](https://docs.fluxcd.io/projects/helm-operator/en/stable/contributing/get-started-developing/) guide
+Refer to the [Get started developing](https://docs.fluxcd.io/en/stable/contributing/get-started-developing/) guide
```
https://docs.fluxcd.io/projects/helm-operator/en/latest/contributing/get-started-developing/

hiddeco commented on README.md:
```diff
@@ -72,7 +72,7 @@ To familiarise yourself with the project and how things work, you might
 
 - [Our contributions guidelines](CONTRIBUTING.md)
-- [Build documentation](https://docs.fluxcd.io/en/stable/contributing/building.html)
+- [Build documentation](https://docs.fluxcd.io/en/stable/contributing/building)
```
https://docs.fluxcd.io/projects/helm-operator/en/latest/contributing/introduction/ or https://docs.fluxcd.io/projects/helm-operator/en/latest/contributing/building/

Commit:
fix links

dholbach commented:
Thanks for the review @hiddeco !
